hawc/apps/epiv2/serializers.py

```diff
@@ -0,0 +1,157 @@
+    class Meta:
+        model = models.AgeProfile
+        fields = ["name"]
+
+    def to_internal_value(self, data):
+        if type(data) is str:
+            try:
+                age_profile = self.Meta.model.objects.get(name__iexact=data)
+                return age_profile
+            except ObjectDoesNotExist:
+                raise serializers.ValidationError(f"'{data}' is not a valid Age profile.")
+
+        return super().to_internal_value(data)
+
+
+class MeasurementTypeSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = models.MeasurementType
+        fields = ["description"]
+
+
+class ChemicalSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = models.Chemical
+        fields = ["id", "name", "dsstox"]
+
+
+class CriteriaSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = models.Criteria
+        fields = ["id", "name"]
+
+
+class ExposureSerializer(serializers.ModelSerializer):
+    measurement_type = MeasurementTypeSerializer(many=True, read_only=True)
+    exposure_route = FlexibleChoiceField(choices=constants.ExposureRoute.choices)
+
+    class Meta:
+        model = models.Exposure
+        exclude = ["design", "created", "last_updated"]
+
+
+class ExposureLevelSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = models.ExposureLevel
+        exclude = ["design", "created", "last_updated"]
+
+
+class OutcomeSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = models.Outcome
+        exclude = ["design", "created", "last_updated"]
+
+
+class AdjustmentFactorSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = models.AdjustmentFactor
+        exclude = ["design", "created", "last_updated"]
+
+
+class DataExtractionSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = models.DataExtraction
+        exclude = ["design", "created", "last_updated"]
+
+
+class DesignSerializer(IdLookupMixin, serializers.ModelSerializer):
+    study = StudySerializer()
+    study_design = FlexibleChoiceField(choices=constants.StudyDesign.choices)
+    source = FlexibleChoiceField(choices=constants.Source.choices)
+    sex = FlexibleChoiceField(choices=constants.Sex.choices)
+    countries = StudyPopulationCountrySerializer(many=True)
+    age_profile = AgeProfileSerializer(many=True)
+    chemicals = ChemicalSerializer(many=True, read_only=True)
+    criteria = CriteriaSerializer(many=True, read_only=True)
+    exposure = ExposureSerializer(many=True, read_only=True)
+    exposure_levels = ExposureLevelSerializer(many=True, read_only=True)
+    outcomes = OutcomeSerializer(many=True, read_only=True)
+    adjustment_factors = AdjustmentFactorSerializer(many=True, read_only=True)
+    data_extractions = DataExtractionSerializer(many=True, read_only=True)
+    url = serializers.CharField(source="get_absolute_url", read_only=True)
+
+    def create(self, validated_data):
```

Comment:
probably best to clean this file up. I'd create a new branch "epi-api-downstream", and i'd keep things in that branch. For this branch, lets' delete them.

That one no content is lost when we're ready to address them, but for now, this is a clean commit that doesn't have non-functional commented out code